x/authz/keeper/keeper.go

```diff
@@ -115,6 +115,14 @@ func (k Keeper) DispatchActions(ctx sdk.Context, grantee sdk.AccAddress, msgs []
 		}
 	}
 
+	// emit the events from the dispatched actions
+	events := msgResult.Events
```

Comment:
So this code needs to be brought inside the loop because now it's just emitting the last msg's events